types/src/contract_event.rs

```diff
@@ -15,12 +15,36 @@ use proptest_derive::Arbitrary;
 use serde::{Deserialize, Serialize};
 use std::{convert::TryFrom, ops::Deref};
 
+pub trait ReadWriteEvent {
+    fn get_event_data(&self) -> (EventKey, u64, &TypeTag, &[u8]);
+    fn update_event_data(&mut self, event_data: &[u8]);
```

Comment:
Why `&[u8]` and not `Vec<u8>` here?

Reply:
The existing method `pub fn event_data(&self) -> &[u8]` is returning `&[u8]`. The `get_event_data` internally invokes the `event_data` method. Do you want me to change both of them to return `Vec<u8>`?

Reply:
I think George asked about update specifically, it makes sense for get to return whatever is appropriate and in particular a slice to avoid copying, but for the update, assuming we don't call update when no modification is necessary, we will have our own data-structure with the updated contents that we can pass, and making that a Vec seems reasonable. So we could pass a Vec<u8> here but keep the slice in event data return type.

Reply:
Yes, I meant update only. For get slice is good.

Reply:
Done.